src/imitation/algorithms/preference_comparisons.py

```diff
@@ -601,6 +610,14 @@ def push(
         self.fragments2.extend(fragments2)
         self.preferences = np.concatenate((self.preferences, preferences))
 
+        # Evict old samples if the dataset is at max capacity
+        if self.max_size is not None:
+            extra = len(self.preferences) - self.max_size
+            if extra > 0:
+                self.fragments1 = self.fragments1[extra:]
```

Comment:
Yeah I thought about using a `collections.deque`, but that doesn't have O(1) random access, which `DataLoader` relies on. `PreferenceDataset.__getitem__()` is getting called `batch_size` times per gradient step, whereas `PreferenceDataset.push()` only gets called once each time we gather new preferences, so it's pretty clear we should optimize for random access. Technically the ideal data structure for this would be a ring buffer, I think, which would get you both properties.